dvc/tree/azure.py

```diff
@@ -132,8 +132,68 @@ def _generate_download_url(self, path_info, expires=3600):
         return blob_client.url + "?" + sas_token
 
     def exists(self, path_info, use_dvcignore=True):
-        paths = self._list_paths(path_info.bucket, path_info.path)
-        return any(path_info.path == path for path in paths)
+        check_path = path_info.path
+        item_exists = False
+        paths = self._list_paths(path_info.bucket, check_path)
+        for path in paths:
```

Comment:
We could just use `funcy.first(paths)` and do the check if result is not None? `break` after one iteration seems unobvious even with the comments. It might be my taste speaking, so its not a blocker.